targets/TARGET_NORDIC/TARGET_NRF5/us_ticker.c

```diff
@@ -494,80 +394,92 @@ static void register_next_tick() {
+{
     nrf_rtc_int_enable(COMMON_RTC_INSTANCE, OS_TICK_INT_MASK);
 
-    nrf_rtc_cc_set(COMMON_RTC_INSTANCE, OS_TICK_CC_CHANNEL, 0);
+    uint32_t current_cnt = nrf_rtc_counter_get(COMMON_RTC_INSTANCE);
+    nrf_rtc_cc_set(COMMON_RTC_INSTANCE, OS_TICK_CC_CHANNEL, current_cnt);
     register_next_tick();
 
-    return nrf_drv_get_IRQn(COMMON_RTC_INSTANCE);
+    NVIC_SetVector(SWI3_IRQn, (uint32_t)SysTick_Handler);
+    NVIC_SetPriority(SWI3_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Emulated Systick Interrupt */
+    NVIC_EnableIRQ(SWI3_IRQn);
+}
+
+// Stop SysTickt timer emulation
+void osRtxSysTimerDisable(void)
+{
+    nrf_rtc_int_disable(COMMON_RTC_INSTANCE, OS_TICK_INT_MASK);
+    
+    // RTC1 is free runing. osRtxSysTimerGetCount will return proper frozen value
+    // thanks to geting frozen value instead of RTC1 counter value
+    frozen_sub_tick = nrf_rtc_counter_get(COMMON_RTC_INSTANCE);
 }
 
+
+
 /**
  * Acknowledge the tick interrupt.
  * This function is called by the function OS_Tick_Handler of RTX.
  * @note this function shouldn't be called directly.
  */
-void os_tick_irqack(void)
+void osRtxSysTimerAckIRQ(void)
 {
-    clear_tick_interrupt();
     register_next_tick();
 }
 
-/**
- * Returns the overflow flag of the alternative hardware timer.
- * @note This function is exposed by RTX kernel.
- * @return 1 if the timer has overflowed and 0 otherwise.
- */
-uint32_t os_tick_ovf(void) {
-    uint32_t current_counter = nrf_rtc_counter_get(COMMON_RTC_INSTANCE);
-    uint32_t next_tick_cc_value = nrf_rtc_cc_get(COMMON_RTC_INSTANCE, OS_TICK_CC_CHANNEL);
-
-    return is_in_wrapped_range(previous_tick_cc_value, next_tick_cc_value, current_counter) ? 0 : 1;
-}
-
-/**
- * Return the value of the alternative hardware timer.
- * @note The documentation is not very clear about what is expected as a result,
- * is it an ascending counter, a descending one ?
- * None of this is specified.
- * The default systick is a descending counter and this function return values in
- * descending order, even if the internal counter used is an ascending one.
- * @return the value of the alternative hardware timer.
- */
-uint32_t os_tick_val(void) {
-    uint32_t current_counter = nrf_rtc_counter_get(COMMON_RTC_INSTANCE);
-    uint32_t next_tick_cc_value = nrf_rtc_cc_get(COMMON_RTC_INSTANCE, OS_TICK_CC_CHANNEL);
-
-    // do not use os_tick_ovf because its counter value can be different
-    if(is_in_wrapped_range(previous_tick_cc_value, next_tick_cc_value, current_counter)) {
-        if (next_tick_cc_value > previous_tick_cc_value) {
-            return next_tick_cc_value - current_counter;
-        } else if(current_counter <= next_tick_cc_value) {
-            return next_tick_cc_value - current_counter;
-        } else {
-            return next_tick_cc_value + (MAX_RTC_COUNTER_VAL - current_counter);
+// provide a free running incremental value over the entire 32-bit range
+uint32_t osRtxSysTimerGetCount(void) {
```

Comment:
@0xc0170 Could you detail the inconsistencies and what you expect ? 
It might be hard to juggle with 3 different coding styles at the same time: 
* Nordic SDK 
* CMSIS RTX
* mbed